code/datums/components/rust.dm

```diff
@@ -19,22 +20,25 @@
 	var/atom/wall_new = new /turf/open/floor/plating(src)
 	wall_new.AddComponent(/datum/component/rust)
 
-/datum/component/rust/Initialize(rust_iconstate = "rust")
+/datum/component/rust/Initialize(iconstate = "rust", icon) // Maybe swap these args?
```

Comment:
At this point, why not just combine them into one? `icon('bla.dmi', "rust")`?

Reply:
If you end up not doing that for whatever reason, use `icon_state` instead of `iconstate` for consistency.

Reply:
I guess you use this for `mutable_appearance` later on, why not just have them pass that directly?